scripts/asm-installer/release_asm_installer

```diff
@@ -118,6 +127,28 @@ get_stable_version() {
   echo "${VER//+/-}"
 }
 
+get_version_file_and_lock() {
+  local FILE_PATH; FILE_PATH=${BUCKET_PATH}/${STABLE_VERSION_FILE}
+
+  if [[ ! $(curl -O "${BUCKET_URL}/${FILE_PATH}") ]]; then
+    echo "Create New Version File ${STABLE_VERSION_FILE}"
+    true >| ${STABLE_VERSION_FILE}
+    gsutil cp "${STABLE_VERSION_FILE}" gs://"${FILE_PATH}"
+  fi
+
+  gsutil retention ${HOLD_TYPE} set "gs://${FILE_PATH}"
+}
+
+upload_version_file_and_unlock() {
+  local FILE_PATH; FILE_PATH=${BUCKET_PATH}/${STABLE_VERSION_FILE}
+
+  sort -r ${STABLE_VERSION_FILE} | uniq > "tmpfile" && mv "tmpfile" ${STABLE_VERSION_FILE}
```

Comment:
This will fail if `tmpfile` exists, and it might not be safe to overwrite it all of the time. Consider

```
TMPFILE="$(mktemp)"
sort ... > "${TMPFILE}" && mv "${TMPFILE}" ...
```

Reply:
Done